core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/storage/FileResolver.scala

```diff
@@ -113,11 +112,12 @@ object FileResolver {
       datasetVersion.getVersionHash
     ) ++ encodedFileRelativePath
 
-    // Build the the format /{did}/{versionHash}/{fileRelativePath}
-    val encodedPath = Paths.get(File.separator, allPathSegments: _*)
+    // Build the the format /{did}/{versionHash}/{fileRelativePath}, both Linux and Windows use forward slash as the splitter
+    val uriSplitter = "/"
+    val encodedPath = uriSplitter + allPathSegments.mkString(uriSplitter)
```

Comment:
can we use some library to do such encoding? https://github.com/lemonlabsuk/scala-uri

Reply:
Seems that this package is more focusing on parsing a given URI. For building a URI, the functionality is essentially the same with the java.URI way.

Reply:
ok thanks